From ecbe591d261516acbee73c1506702bc78f9a90cc Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dimitar@dinux.eu>
Date: Sun, 16 Oct 2016 22:27:25 +0300
Subject: [PATCH 12/23] ld: testsuite: Fix srec test setup for PRU

2016-10-16  Dimitar Dimitrov <dimitar@dinux.eu>

	* ld/testsuite/ld-srec/srec.exp (run_srec_test): Fix srec test setup
	for PRU.

Signed-off-by: Dimitar Dimitrov <dimitar@dinux.eu>
---
 ld/testsuite/ld-srec/srec.exp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ld/testsuite/ld-srec/srec.exp b/ld/testsuite/ld-srec/srec.exp
index 9fcf07a..10b0a51 100644
--- a/ld/testsuite/ld-srec/srec.exp
+++ b/ld/testsuite/ld-srec/srec.exp
@@ -300,6 +300,11 @@ proc run_srec_test { test objs } {
 	set flags "$flags -no-relax"
     }
 
+    # PRU ELF target relaxes by default; S-Record linker does not
+    if [istarget pru*-*-*] {
+	set flags "$flags -no-relax"
+    }
+
     if { ![ld_simple_link $ld tmpdir/sr1 "$flags $objs"] \
 	 || ![ld_simple_link $ld tmpdir/sr2.sr "$flags --oformat srec $objs"] } {
 	fail $test
-- 
2.10.2

